libdrgn/symbol.c

```diff
@@ -40,6 +45,39 @@ void drgn_symbol_from_elf(const char *name, uint64_t address,
 		ret->kind = DRGN_SYMBOL_KIND_UNKNOWN;
 }
 
+LIBDRGN_PUBLIC void drgn_symbol_source(struct drgn_symbol *sym,
+				       struct drgn_program *prog,
+				       unsigned long offset,
+				       size_t size,
+				       char *src_file_ret,
+				       int *line_num_ret,
+				       int *line_col_ret)
+{
+	if (sym->kind != DRGN_SYMBOL_KIND_FUNC)
+		return;
```

Comment:
Needs an error return, otherwise caller won't know if `src_file_ret` is a valid string.